records_versioning/services.py

```
@@ -0,0 +1,255 @@
+    - تابع مستقیماً از ORM برای بارگذاری مدل و نسخه‌ها استفاده می‌کند؛ 
+      در صورت وجود نداشتن مدل یا نسخهٔ درخواستی، استثناهای استاندارد ORM 
+      (مثل DoesNotExist) از سمت فراخوانده‌کننده مدیریت می‌شوند.
+    """
+    # Input validation
+    if not isinstance(target_version, int) or target_version <= 0:
+        raise ValueError("target_version must be a positive integer")
+    
+    try:
+        app_label, model_name = RESOURCE_MAP[resource_type]
+    except KeyError:
+        raise ValueError(f"Unknown resource_type: {resource_type}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)